src/primitives/sphere.jl

```diff
@@ -37,6 +37,36 @@ end
 
 Sphere(p1::Tuple, p2::Tuple, p3::Tuple) = Sphere(Point(p1), Point(p2), Point(p3))
 
+"""
+    Sphere(p1, p2, p3, p4)
+
+A 3D sphere passing through points `p1`, `p2`, `p3` and `p4`.
+"""
+function Sphere(p1::Point{3}, p2::Point{3}, p3::Point{3}, p4::Point{3})
+  v1 = coordinates(p1)
+  v2 = coordinates(p2)
+  v3 = coordinates(p3)
+  v4 = coordinates(p4)
+  M = hcat(v1, v2, v3, v4)
+  a = 2 * det(vcat(M, ones((1, 4))))
+  if a == 0
```

Comment:
Equality with floating point is not ideal. Maybe you meant isapprox(a, zero(T), atol=atol(T))

Reply:
What should I take for `T` here?
And are you sure `isapprox(x, 0)` is reliable? I vaguely remember there is a problem when comparing with zero, but I don't remember, and I'm not sure.

Reply:
T is the coordtype(p1) for example. It is usually Float64 or Float32 in our tests.

isapprox is not ideal either, but better than == at least. We will consider exact coordinates at some point in the future as well, but for now we are using floating point.

Reply:
Ah yes, `atol` is absolute tolerance, this should be nice.

Reply:
What if `coordtype(p2)` is different from `coordtype(p1)`? Isn't it better to take `T = typeof(a)`?

Reply:
That sounds good too